File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int pi=0;
        int ni=1;

        vector<int> ans(nums.size(),0);

        for(int i:nums)
        {
            if(i<0)
            {
                ans[ni]=i;
                ni+=2;
            }
            else
            {
                ans[pi]=i;
                pi+=2;
            }
        }
        return ans;
    }
};

File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        ans=[0]*len(nums)
        pi=0
        ni=1

        for i in nums:
            if i<0:
                ans[ni]=i
                ni+=2
            else:
                ans[pi]=i
                pi+=2
        return ans